packages/source-iotsitewise/src/time-series-data/subscribeToTimeSeriesData.ts

```diff
@@ -23,17 +30,83 @@ export const subscribeToTimeSeriesData =
 
     const errors: Record<string, ErrorDetails> = {};
 
+    let annotations: Annotations = {};
+
+    const alarms: AlarmsStore = {};
+
+    queries.forEach(({ assets }) =>
+      assets.forEach(({ assetId, properties }) =>
+        properties.forEach(({ alarms }) => {
+          if (alarms) {
+            // TODO: get all the alarms where inputProperty = correspondingQuery.propertyId
+            /**
+             * iterate over all compositeModels that are type AWS/ALARM and get the id of AWS/ALARM_STATE property
+             * describe the alarm using alarmModule and find if the inputPropertyId = propertyId
+             *
+             * override queries with queries with alarm streams
+             */
+          }
+        })
+      )
+    );
+
     const emit = () => {
       callback({
-        dataStreams: completeDataStreams({ dataStreams, assetModels }),
+        dataStreams: completeDataStreams({ dataStreams, assetModels, alarms }),
         viewport,
+        annotations,
       });
     };
 
     const { update, unsubscribe } = dataModule.subscribeToDataStreams({ queries, request }, (data) => {
       dataStreams = data.dataStreams;
       viewport = data.viewport;
       emit();
+
+      annotations.y = [];
+      Object.keys(assetModels).forEach(assetId => {
+        const assetModel = assetModels[assetId];
+
```

Comment:
a more broad comment: the subscribeToTimeSeriesData method is where a bunch of stuff gets tied together and its getting pretty complicated, if there's anyway to move complexity away from here / encapsulate it elsewhere it hink that would be best

Reply:
Agreed, will work to find a better solution.